airflow/dag/serialization/json_schema.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""jsonschema for validating serialized DAG and operator."""
+
+from typing import Optional
+
+from airflow.dag.serialization.enum import DagAttributeTypes as DAT, Encoding
+
+
+def make_object_schema(
+        var_schema: Optional[dict] = None, type_enum: Optional[list] = None) -> dict:
+    """jsonschema of an encoded object."""
+    schema = {
+        'type': 'object',
+        'properties': {
+            Encoding.TYPE.value: {'type': 'string'}
+        },
+        'required': [
+            Encoding.TYPE.value,
+            Encoding.VAR.value
+        ]
+    }
+    if var_schema is not None:
+        schema['properties'][Encoding.VAR.value] = var_schema
+
+    if type_enum is not None:
+        schema['properties'][Encoding.TYPE.value]['enum'] = type_enum
```

Comment:
Is this correct?

Or should it be:

```suggestion
        schema['properties'][Encoding.TYPE.value] = type_enum
```